Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -185,26 +189,86 @@ class ContactViewController: OWSViewController, CNContactViewControllerDelegate
+        bottomView.autoPinEdge(.top, to: .bottom, of: topView)
+        bottomView.autoPinEdge(toSuperviewEdge: .bottom)
+
+        let scrollView = UIScrollView()
+        scrollView.preservesSuperviewLayoutMargins = false
+        self.view.addSubview(scrollView)
+        scrollView.layoutMargins = .zero
+        scrollView.autoPinWidthToSuperview()
+        scrollView.autoPinEdge(.top, to: .bottom, of: topView)
+        scrollView.autoPinEdge(toSuperviewEdge: .bottom)
+
+        let fieldsView = createFieldsView()
+
+        // See notes on how to use UIScrollView with iOS Auto Layout:
+        //
+        // https://developer.apple.com/library/content/releasenotes/General/RN-iOSSDK-6_0/
+        scrollView.addSubview(fieldsView)
+        fieldsView.autoPinLeadingToSuperviewMargin()
+        fieldsView.autoPinTrailingToSuperviewMargin()
+        fieldsView.autoPinEdge(toSuperviewEdge: .top)
+        fieldsView.autoPinEdge(toSuperviewEdge: .bottom)
+    }
+
+    private func createTopView() -> UIView {
+        SwiftAssertIsOnMainThread(#function)
+
         let topView = UIView.container()
         topView.backgroundColor = UIColor(rgbHex: 0xefeff4)
-        topView.preservesSuperviewLayoutMargins = true
-        rootView.addSubview(topView)
-        topView.autoPinEdge(toSuperviewEdge: .top)
-        topView.autoPinEdge(.left, to: .left, of: self.view)
-        topView.autoPinEdge(.right, to: .right, of: self.view)
+        topView.preservesSuperviewLayoutMargins = false
+
+        // Back Button
+        let backButtonSize = CGFloat(50)
+        let backButton = TappableView(actionBlock: { [weak self] _ in
+            guard let strongSelf = self else { return }
+            strongSelf.didPressDismiss()
+        })
+        backButton.autoSetDimension(.width, toSize: backButtonSize)
+        backButton.autoSetDimension(.height, toSize: backButtonSize)
+        topView.addSubview(backButton)
+        backButton.autoPin(toTopLayoutGuideOf: self, withInset: 0)
+        backButton.autoPinLeadingToSuperviewMargin()
+
+        let backIconName = (self.view.isRTL() ? "system_disclosure_indicator" : "system_disclosure_indicator_rtl")
+        let backIconImage = UIImage(named: backIconName)?.withRenderingMode(.alwaysTemplate)
+        let backIconView = UIImageView(image: backIconImage)
+        backIconView.contentMode = .scaleAspectFit
+        backIconView.tintColor = UIColor.black.withAlphaComponent(0.6)
+        backButton.addSubview(backIconView)
+        backIconView.autoCenterInSuperview()
 
         // TODO: Use actual avatar.
         let avatarSize = CGFloat(100)
-        let avatarView = UIView.container()
-        avatarView.backgroundColor = UIColor.ows_materialBlue
-        avatarView.layer.cornerRadius = avatarSize * 0.5
+
+        let avatarView = AvatarImageView()
+        // TODO: What's the best colorSeed value to use?
+        let avatarBuilder = OWSContactAvatarBuilder(nonSignalName: contact.displayName,
```

Comment:
Use proper default avatars.

Reply:
I like this idea.  I'm going to address it in a subsequent PR.  There's upcoming e164/phone parsing work to be done that this will fit with.